src/services/DfusionService.ts

```diff
@@ -77,6 +81,43 @@ export class DfusionRepoImpl implements DfusionService {
     this._web3 = web3
   }
 
+  public async isHealthy (): Promise<boolean> {
+    try {
+      // Try to get info from the node, and the peer count, and the last mined block
+      await this._web3.eth.getNodeInfo()
+      const peerCount = await this._web3.eth.net.getPeerCount()
```

Comment:
Another good check is `await web3.eth.net.isListening()`

Reply:
8ffa73e1b51b44bd9069be6d1268ab28e1008012